packages/react-vapor/src/components/sideNavigation/SideNavigationMenuSection.tsx

```diff
@@ -1,63 +1,86 @@
 
-export interface ISideNavigationSectionOwnProps {
+export interface ISideNavigationSectionProps extends SideNavigationHeaderProps {
+    /**
+     * @deprecated pass those in props directly
+     */
     header?: ISideNavigationHeaderProps;
     expandable?: boolean;
-}
-
-export interface ISideNavigationSectionStateProps {
     expanded?: boolean;
-}
-
-export interface ISideNavigationSectionDispatchProps {
     onClick?: () => void;
 }
 
-export interface ISideNavigationSectionProps
-    extends ISideNavigationSectionOwnProps,
-        ISideNavigationSectionStateProps,
-        ISideNavigationSectionDispatchProps {}
-
-const arrowClassName = 'collapsible-arrow icon fill-white transparency-4';
-
-export class SideNavigationMenuSection extends React.Component<ISideNavigationSectionProps> {
-    private handleClick() {
-        if (this.props.onClick) {
-            this.props.onClick();
-        }
-    }
+export interface SideNavigationHeaderProps {
+    title?: React.ReactNode;
+    svgName?: string;
+    svgClass?: string;
+    customIcon?: React.ReactNode;
+    onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+}
 
-    render() {
-        return (
-            <div className="block navigation-menu-section">
-                {this.buildHeader()}
-                <SlideY in={!(this.props.expandable && !this.props.expanded)} timeout={350}>
-                    <div className="navigation-menu-section-items">{this.props.children}</div>
-                </SlideY>
-            </div>
+const SideNavigationHeader: React.FunctionComponent<SideNavigationHeaderProps> = (props) => {
+    let icon: React.ReactNode = null;
+    if (props.customIcon) {
+        icon = props.customIcon;
+    } else if (props.svgName) {
+        icon = (
+            <Svg
+                svgName={props.svgName}
+                svgClass={classNames(
+                    'navigation-menu-section-header-icon icon mod-lg transparency-3 fill-white',
+                    props.svgClass
+                )}
+            />
         );
+    } else {
+        icon = <span className="navigation-menu-section-header-icon" />;
     }
 
-    private buildHeader(): JSX.Element {
-        if (this.props.header) {
-            let arrow = null;
-            if (this.props.expandable) {
-                arrow = this.props.expanded ? (
-                    <Svg svgName="arrow-top-rounded" className={arrowClassName} />
-                ) : (
-                    <Svg svgName="arrow-bottom-rounded" className={arrowClassName} />
-                );
-            }
-            return (
-                <SideNavigationHeader {...this.props.header} onClick={() => this.handleClick()}>
-                    {arrow}
-                </SideNavigationHeader>
-            );
-        }
-        return <SideNavigationLoadingHeader />;
-    }
-}
+    return !_.isEmpty(React.Children.map(props.children, _.identity)) ? (

```

Comment:
what is this sorcery? :P

Looks like a `hasChildren`? Can it be named as such?

On a higher level, why is this check required at all?

Reply:
This high Griffindor sorcery is indeed `hasChildren`.

The use case is when the side navigation section has no title. In that situation, you don't want to display a header with an empty content, you simply don't want to display the header at all. 

It visually looks like this:
![image](https://user-images.githubusercontent.com/35579930/63376286-75f5e400-c35b-11e9-93c3-f4e427303e74.png)

I can simply verify if the title is empty a little earlier and if so not render a `SideNavigationHeader` at all. Would that be simpler?